src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2596,6 +2602,19 @@ void TorrentImpl::updateProgress()
     }
 }
 
+void TorrentImpl::updateMaxConnections()
+{
+    const int maxDownloadingConnections = m_session->maxConnectionsPerDownloadingTorrent();
+    const int maxSeedingConnections = m_session->maxConnectionsPerSeedingTorrent();
+    const bool useSeedingLimit = (maxSeedingConnections != -1) && isUploading();
+    const int max = useSeedingLimit ? maxSeedingConnections : maxDownloadingConnections;
+
+    if (nativeHandle().max_connections() == max)
```

Comment:
It is a bad idea. `torrent_handle::max_connections()` is blocking so you will not optimize the performance in such a way.